src/agent/rustjail/src/validator.rs

```diff
@@ -58,10 +59,10 @@ fn rootfs(root: &str) -> Result<()> {
         cleaned.push(e);
     }
 
-    let canon = path.canonicalize().context("canonicalize")?;
+    let canon = path.canonicalize().context("failed to canonicalize path")?;
     if cleaned != canon {
         // There is symbolic in path
-        return Err(einval());
+        return Err(anyhow!("Cleaned and canonicalized path do not match. There may be illegal symbols in the path name"));
```

Comment:
- The invalid path isn't being logged in the message so this is reliant on the caller logging it. Could you check/adjust as necessary?
- This canonicalisation check is another one that could be added to a new `check_path()` function possibly.

Reply:
I was working on a different [issue](https://github.com/kata-containers/kata-containers/issues/1378) converting String to OsString/Paths. While I haven't look at this too closely, it looks like this is making sure that paths are valid Strings. Perhaps we might not need this check in the future if we are able to properly replace String instances?